src/PhpSpreadsheet/Calculation/TextData/Extract.php

```diff
@@ -26,7 +26,7 @@ public static function left($value = '', $chars = 1): string
             $value = ($value) ? Calculation::getTRUE() : Calculation::getFALSE();
         }
 
-        return mb_substr($value, 0, $chars, 'UTF-8');
+        return mb_substr($value ?? '', 0, $chars, 'UTF-8');
```

Comment:
In the case of the Excel function implementations (particularly in the statistical functions), I typically handled these "optional" values in the initial extraction of values:

```php
public static function left($value = '', $chars = 1): string
{
    $value = ($value === null) ? '' : Functions::flattenSingleValue($value);
    $chars = ($chars === null) ? 1 : Functions::flattenSingleValue($chars);
    ...
}
```
Perhaps a similar approach would help create more consistency in the code

Reply:
For consistency's sake, I think these functions should use the same model as Engineering, MathTrig, DateTime, etc. - put the flattens and validations inside a try-catch block. I would be willing to look at this for all of TextData, but not as part of this PR.

My attempt to do something like this resulted in test failures, because your recoding above isn't quite right. Omitting the chars parameter from LEFT does act as if chars was supplied as one. However, supplying a null (e.g. `LEFT("ABC", A2)`, where A2 is an empty cell) makes LEFT act as if chars was supplied as zero. It is easy to fix your sample to reflect this, but I suspect there may be similar problems in other TextData functions. Again, I would be willing to investigate further, but not as part of this PR.

Also, the chars parameter is optional for LEFT and RIGHT, but chars and start are both required for MID, which the existing code does not recognize.

Reply:
Excel is a PITA sometimes, particularly with the way it allows "missing" arguments, that aren't trailing values.... the calculation engine passes a null in those cases

Now that we've done the bulk work on moving the Excel functions out of the mega classes, it will be a lot more practical to rationalise all validations down to a common validation class; and apply the flattens within the try/catch as you suggest
Further investigation could be awkward, as I'm not sure that OpenOffice and Gnumeric necessarily follow Excel's behaviour; and I don't believe that Excel itself has been completely consistent from version to version